tests/unit/test_charm_configure.py

```diff
@@ -0,0 +1,525 @@
+    )
+    patcher_webui_url = patch(
+        "charms.sdcore_nms_k8s.v0.sdcore_config.SdcoreConfigRequires.webui_url",
+        new_callable=PropertyMock,
+    )
+    patcher_is_resource_created = patch(
+        "charms.data_platform_libs.v0.data_interfaces.DatabaseRequires.is_resource_created"
+    )
+    patcher_get_assigned_certificate = patch(
+        "charms.tls_certificates_interface.v4.tls_certificates.TLSCertificatesRequiresV4.get_assigned_certificate"
+    )
+    patcher_db_fetch_relation_data = patch(
+        "charms.data_platform_libs.v0.data_interfaces.DatabaseRequires.fetch_relation_data"
+    )
+
+    @pytest.fixture(autouse=True)
+    def context(self):
+        self.ctx = scenario.Context(
+            charm_type=AMFOperatorCharm,
+        )
+
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        self.mock_get_assigned_certificate = (
+            TestCharmConfigure.patcher_get_assigned_certificate.start()
+        )
+        self.mock_is_resource_created = TestCharmConfigure.patcher_is_resource_created.start()
+        self.mock_nrf_url = TestCharmConfigure.patcher_nrf_url.start()
+        self.mock_webui_url = TestCharmConfigure.patcher_webui_url.start()
+        self.mock_check_output = TestCharmConfigure.patcher_check_output.start()
+        self.mock_k8s_service = TestCharmConfigure.patcher_k8s_service.start().return_value
+        self.mock_db_fetch_relation_data = (
+            TestCharmConfigure.patcher_db_fetch_relation_data.start()
+        )
+
+    @staticmethod
+    def _read_file(path: str) -> str:
+        """Read a file and returns as a string.
+
+        Args:
+            path (str): path to the file.
+
+        Returns:
+            str: content of the file.
+        """
+        with open(path, "r") as f:
+            content = f.read()
+        return content
+
+    def example_cert_and_key(self, tls_relation_id: int) -> tuple[ProviderCertificate, PrivateKey]:
+        private_key_str = generate_private_key()
+        csr = generate_csr(
+            private_key=private_key_str,
+            common_name="amf",
+        )
+        ca_private_key = generate_private_key()
+        ca_certificate = generate_ca(
+            private_key=ca_private_key,
+            common_name="ca.com",
+            validity=365,
+        )
+        certificate_str = generate_certificate(
+            csr=csr,
+            ca=ca_certificate,
+            ca_key=ca_private_key,
+            validity=365,
+        )
+        provider_certificate = ProviderCertificate(
+            relation_id=tls_relation_id,
+            certificate=Certificate.from_string(certificate_str),
+            certificate_signing_request=CertificateSigningRequest.from_string(csr),
+            ca=Certificate.from_string(ca_certificate),
+            chain=[Certificate.from_string(ca_certificate)],
+        )
+        private_key = PrivateKey.from_string(private_key_str)
+        return provider_certificate, private_key
```

Comment:
I think this method is duplicated in different test files, could it be moved to another file like `conftest.py` to remove the duplication?

Reply:
Agreed, done